### ISSUE TITLE
Hyper-Parameter settings on ImageNet

### ISSUE THREAD

**Issue by Noah-DL** [PeterouZh/Omni-GAN-PyTorch#2]
Hi, I am reproducing the results of Omni-GAN on ImageNet128. I used the same settings as BigGAN except I changed the batch size, added weight decay to Generator and Discriminator. But the generated images occured mode collapse at very early training stage. Below is the scripts I used.

_python3 train.py \
--dataset I128_hdf5 --parallel --shuffle  --num_workers 8 --batch_size 256 \
--num_G_accumulations 1 --num_D_accumulations 1 --load_in_mem \
--num_D_steps 1 --G_lr 1e-4 --D_lr 4e-4 --D_B2 0.999 --G_B2 0.999 \
--G_attn 64 --D_attn 64 --G_nl inplace_relu --D_nl inplace_relu \
--SN_eps 1e-6 --BN_eps 1e-5 --adam_eps 1e-6 --num_epochs 200 \
--G_ortho 0.0 --G_shared --G_init ortho --D_init ortho \
--hier --dim_z 120 --shared_dim 128 --G_eval_mode --G_ch 96 --D_ch 96 \
--ema --use_ema --ema_start 20000 --use_multiepoch_sampler \
--data_root ../data --sv_log_interval -1 \
--test_every 2000 --save_every 1000 --num_best_copies 1 --num_save_copies 1 --seed 0 \
--loss_type omni --G_WD 1e-5 --D_WD 1e-5_

I also checked experiments with projection loss and without weight decay and reset the number of accumulations to 8 (batch size equals to 2048), the resulting images were normal.

Are there any differences between my settings and yours? Could you give me some suggestions. By the way, what's the batch size you use for ImageNet 32x32? Do you use batch size of 32? it seems very small and the networks will need more time to update.

**Comment by PeterouZh** [comment]
Hi, 

The omni-loss is a much **stronger** classification loss than the projection-based loss and cannot employ a large batch size similar to that of the BigGAN. For the **ImageNet128x128** experiment, the core hyperparameters are modified as follows:

```
batch_size: 256
num_G_accumulations: 1
num_D_accumulations: 1
num_D_steps: 1
Generator:
    weight_decay: 0.001
Discriminator:
    weight_decay: 0.00001
```
However, the script you provided above seems to be consistent with the hyperparameters I used, and I don't know why you didn't get good results. By the way, I used 8*v100 with 32G VRAM each for the ImageNet 128x128 experiment. 

As for the ImageNet 32x32 experiment shown in the paper, since I didn't have large-scale GPU resources at that time, the experiments in the paper were conducted on a single 1080Ti GPU. The batch size is set to 64 for ImageNet 32x32. 

I'm curious why your experiment didn't work well. I provide the more detailed configuration as follows:
```
  args:
    dataset: I128_hdf5
    data_root: "datasets/ImageNet_hdf5/ILSVRC128.hdf5"
    num_workers: 8
    batch_size: 256
    num_G_accumulations: 1
    num_D_accumulations: 1
    num_D_steps: 1
    G_lr: 1e-4
    D_lr: 4e-4
    D_B2: 0.999
    G_B2: 0.999
    G_attn: '64'
    D_attn: '64'
    G_nl: inplace_relu
    D_nl: inplace_relu
    SN_eps: 1e-6
    BN_eps: 1e-5
    adam_eps: 1e-6
    G_ortho: 0.0
    G_shared: true
    G_init: ortho
    D_init: ortho
    hier: true
    dim_z: 120
    shared_dim: 128
    G_eval_mode: true
    G_ch: 96
    D_ch: 96
    ema: true
    use_ema: true
    ema_start: 20000
    test_every: 2000
    save_every: 1000
    num_best_copies: 4
    num_save_copies: 2
    seed: 0
    use_multiepoch_sampler: true
  Discriminator:
    output_dim: 1002
    weight_decay: 0.00001
  Generator:
    weight_decay: 0.001
  omni_loss:
    mode: p_and_n
    gamma: 1.0
    margin: 0.0
```
I hope this helps!

**Comment by Noah-DL** [comment]
Thank you very much for your reply, let me try it again.

**Comment by PeterouZh** [comment]
If you have any questions, please let me know. Thank you！

**Comment by PeterouZh** [comment]
I am very sad that the current review of our paper is not so good. Thus I decide not to disclose the details of the experiment on ImageNet 128 x 128. All code and hyperparameters for the ImageNet 128/256 experiment will not be made public unless our paper is accepted. Thanks for your interest.

**Comment by Noah-DL** [comment]
I can feel what you said. No worries. I believe your work is a great job. Hope your paper can be accepted.

**Comment by PeterouZh** [comment]
> I can feel what you said. No worries. I believe your work is a great job. Hope your paper can be accepted.

Thank you so much for your encouragement.
